docs/security.md

```diff
@@ -15,48 +15,73 @@ Follow these instructions to [authenticate in strict mode](https://docs.mesosphe
 
-Here are the required variables:
-
-| Variable                   | Description                                     |
-|----------------------------|-------------------------------------------------|
-| `spark.ssl.enabled`        | Whether to enable SSL (default: `false`).       |
-| `spark.ssl.keyStoreBase64` | Base64 encoded blob containing a Java keystore. |
 
 The Java keystore (and, optionally, truststore) are created using the [Java keytool][12]. The keystore must contain one private key and its signed public key. The truststore is optional and might contain a self-signed root-ca certificate that is explicitly trusted by Java.
 
-Both stores must be base64 encoded, for example:
+Both stores must be base64 encoded without newlines, for example:
 
-    cat keystore | base64 /u3+7QAAAAIAAAACAAAAAgA...
+```bash
+cat keystore | base64 -w 0 > keystore.base64
+cat keystore.base64
+/u3+7QAAAAIAAAACAAAAAgA...
+```
 
 **Note:** The base64 string of the keystore will probably be much longer than the snippet above, spanning 50 lines or so.
 
-With this and the password `secret` for the keystore and the private key, your JSON options file will look like this:
-
-    {
-      "security": {
-        "ssl": {
-          "enabled": true,
-          "keyStoreBase64": "/u3+7QAAAAIAAAACAAAAAgA...”,
-          "keyStorePassword": "secret",
-          "keyPassword": "secret"
-        }
-      }
-    }
-
-Install Spark with your custom configuration:
-
-    dcos package install --options=options.json spark
-
-Make sure to connect the DC/OS cluster only using an SSL connection (i.e. by using an `https://<dcos-url>`). Use the following command to set your DC/OS URL:
+Add the stores to your secrets in the DC/OS secret store. For example, if your base64-encoded keystores 
+and truststores are server.jks.base64 and trust.jks.base64, respectively, then use the following 
+commands to add them to the secret store: 
+
+```bash
+dcos security secrets create /__dcos_base64__truststore --value-file trust.jks.base64
+dcos security secrets create /__dcos_base64__keystore --value-file server.jks.base64
+```
+
+In this case, you are adding two secrets `/truststore` and `/keystore` that you will need to pass to the Spark Driver and Executors. 
+You must add the following configurations to your `dcos spark run ` command:
+
+```bash
+
+dcos spark run --verbose --submit-args="\
+--conf spark.mesos.containerizer=mesos \  # use mesos containerizer
+--conf spark.ssl.enabled=true \
+--conf spark.ssl.enabledAlgorithms=TLS_RSA_WITH_AES_128_CBC_SHA256 \
+--conf spark.ssl.keyPassword=<key password> \
+--conf spark.ssl.keyStore=server.jks \  # This MUST be set this way
+--conf spark.ssl.keyStorePassword=<keystore access password> \
+--conf spark.ssl.protocol=TLS \
+--conf spark.ssl.trustStore=trust.jks \  # this MUST be set this way
+--conf spark.ssl.trustStorePassword=<truststore password> \
+--conf spark.mesos.driver.secret.names=__dcos_base64__keystore,__dcos_base64__truststore \
```

Comment:
@keithchambers should probably review this. We could add some sugar to the CLI so these extra configs aren't required (like we do with Kerberos), but it becomes the same thing.